OMEdit/OMEditGUI/Animation/AbstractAnimationWindow.cpp

```diff
@@ -207,9 +213,138 @@ void AbstractAnimationWindow::createActions()
 
 /*!
+ * \brief AbstractAnimationWindow::initInteractiveControlPanel
+ */
+void AbstractAnimationWindow::initInteractiveControlPanel()
+{
+  if (getVisualizer()) {
+    VisualizerFMU* FMUvis = dynamic_cast<VisualizerFMU*>(mpVisualizer);
+    QWidget *widget = new QWidget(this);
+    mSpinBoxVector.clear();
+    mStateLabels.clear();
+
+    if (FMUvis->getFMU()->getFMUData()->_stateNames.size()==FMUvis->getFMU()->getFMUData()->_nStates){
+    //widgets for the states
+    QVBoxLayout *layout = new QVBoxLayout();
+    layout->addWidget(new QLabel(QString("<b>modify state variables</b>"),this));
+    for (unsigned int stateIdx = 0; stateIdx < FMUvis->getFMU()->getFMUData()->_nStates; stateIdx++)
+    {
+      DoubleSpinBoxIndexed* spinBox = new DoubleSpinBoxIndexed(this, stateIdx);
+      spinBox->setValue(FMUvis->getFMU()->getFMUData()->_states[stateIdx]);
+      spinBox->setMaximum(DBL_MAX);
+      spinBox->setMinimum(-DBL_MAX);
+      spinBox->setSingleStep(0.1);
+      mSpinBoxVector.push_back(spinBox);
+
+      QLabel* stateLabel = new QLabel(QString::number(FMUvis->getFMU()->getFMUData()->_states[stateIdx]), this);
+      mStateLabels.push_back(stateLabel);
+
+      QWidget* stateValWidget = new QWidget();
+      QHBoxLayout *stateInfoLayout = new QHBoxLayout(this);
+      stateInfoLayout->addWidget(stateLabel);
+      stateInfoLayout->addWidget(spinBox);
+      stateValWidget->setLayout(stateInfoLayout);
+
+      layout->addWidget(new QLabel(QString::fromStdString(FMUvis->getFMU()->getFMUData()->_stateNames.at(stateIdx)),this));
+      layout->addWidget(stateValWidget);
+      connect(mSpinBoxVector.at(stateIdx), SIGNAL(valueChangedFrom(double, int)), this, SLOT(setStateSolveSystem(double, int)));
+    }
+
+    //widgets for the inputs
+    //layout->addWidget(new QLabel(QString("<b>modify inputs</b>"),this));
+    for (int inputIdx = 0; inputIdx < 0; inputIdx++) {
+      std::string name = "some input";
+      layout->addWidget(new QLabel(QString::fromStdString("inputName"),this));
+      QWidget* valWidget = new QWidget();
+      QHBoxLayout *valueLayout = new QHBoxLayout(this);
+      valueLayout->addWidget(new QLabel(QString::fromStdString("min"),this));
+      valueLayout->addWidget(new QLabel(QString::fromStdString("value"),this));
+      valueLayout->addWidget(new QLabel(QString::fromStdString("max"),this));
+      valWidget->setLayout(valueLayout);
+      layout->addWidget(valWidget);
+      layout->addWidget(new QSlider(Qt::Horizontal, this));
+    }
+    /*
+    QPushButton* updateButton = new QPushButton(QIcon(":/Resources/icons/update.svg"), "&update system",this);
+    layout->addWidget(updateButton);
+    connect(updateButton, SIGNAL(released()), this, SLOT(setStatesSolveSystem()));
+    */
+    layout->setSizeConstraint(QLayout::SetMinimumSize);
+    widget->setLayout(layout);
+    QScrollArea *scrollArea = new QScrollArea(this);
+    scrollArea->setWidget(widget);
+    mpAnimationParameterDockerWidget->setWidget(scrollArea);
```

Comment:
The whole layout of `Parameter Settings` is not nifty. Maybe you can play with layout alignment, spacing and margin to get better results.

Its not necessary to fix this, just my personal suggestion.